source/containers.tex

```diff
@@ -16086,6 +16093,1215 @@
+    template<class K> bool contains(const K& x) const;
+
+    iterator lower_bound(const key_type& x);
+    const_iterator lower_bound(const key_type& x) const;
+    template<class K> iterator lower_bound(const K& x);
+    template<class K> const_iterator lower_bound(const K& x) const;
+
+    iterator upper_bound(const key_type& x);
+    const_iterator upper_bound(const key_type& x) const;
+    template<class K> iterator upper_bound(const K& x);
+    template<class K> const_iterator upper_bound(const K& x) const;
+
+    pair<iterator, iterator> equal_range(const key_type& x);
+    pair<const_iterator, const_iterator> equal_range(const key_type& x) const;
+    template<class K>
+      pair<iterator, iterator> equal_range(const K& x);
+    template<class K>
+      pair<const_iterator, const_iterator> equal_range(const K& x) const;
+
+    friend bool operator==(const flat_set& x, const flat_set& y);
+
+    friend @\exposconcept{synth-three-way-result}@<value_type>
+    operator<=>(const flat_set& x, const flat_set& y);
+
+    friend void swap(flat_set& x, flat_set& y) noexcept { x.swap(y); }
+
+  private:
+    container_type @\exposid{c}@;           // \expos
+    key_compare @\exposid{compare}@;        // \expos
+  };
+
+  template<class InputIterator, class Compare = less<@\placeholder{iter-value-type}@<InputIterator>>>
+    flat_set(InputIterator, InputIterator, Compare = Compare())
+      -> flat_set<@\placeholder{iter-value-type}@<InputIterator>, Compare>;
+
+  template<class InputIterator, class Compare = less<@\placeholder{iter-value-type}@<InputIterator>>>
+    flat_set(sorted_unique_t, InputIterator, InputIterator, Compare = Compare())
+      -> flat_set<@\placeholder{iter-value-type}@<InputIterator>, Compare>;
+
+  template<ranges::input_range R, class Compare = less<ranges::range_value_t<R>>,
+          class Allocator = allocator<ranges::range_value_t<R>>>
+    flat_set(from_range_t, R&&, Compare = Compare(), Allocator = Allocator())
+      -> flat_set<ranges::range_value_t<R>, Compare>;
+
+   template<ranges::input_range R, class Allocator>
+     flat_set(from_range_t, R&&, Allocator)
+       -> flat_set<ranges::range_value_t<R>, less<ranges::range_value_t<R>>>;
+
+  template<class Key, class Compare = less<Key>>
+    flat_set(initializer_list<Key>, Compare = Compare())
+      -> flat_set<Key, Compare>;
+
+  template<class Key, class Compare = less<Key>>
+    flat_set(sorted_unique_t, initializer_list<Key>, Compare = Compare())
+      -> flat_set<Key, Compare>;
+
+  template<class Key, class Compare, class KeyContainer, class Allocator>
+    struct uses_allocator<flat_set<Key, Compare, KeyContainer>, Allocator>
+      : bool_constant<uses_allocator_v<KeyContainer, Allocator>> { };
+}
+\end{codeblock}
+
+\rSec3[flat.set.ctor]{Constructors}
+
+\indexlibraryctor{flat_set}%
+\begin{itemdecl}
+flat_set(container_type cont);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{c} with \tcode{std::move(cont)},
```

Comment:
Replace all `\exposid{c}` with `\tcode{c}`.

Reply:
I think these are implementation names.

Reply:
We should look at this again later, there definitely seems to be cause for some cleanup.